src/lib-sieve/sieve-script.c

```diff
@@ -748,7 +748,7 @@ int sieve_script_binary_save_default(struct sieve_script *script ATTR_UNUSED,
 	if (storage->bin_path != NULL &&
 	    str_begins_with(path, storage->bin_path) &&
 	    sieve_storage_setup_bin_path(
-		script->storage, mkdir_get_executable_mode(save_mode)) < 0)
+		script->storage, 0700) < 0)
```

Comment:
I wonder if we should instead create a new mkdir_get_wx_mode() that adds +w and +x bits for every +r. @stephanbosch do you remember if there's a reason we want to get the mode from the sieve script rather than hardcoding to 0700?